### ISSUE TITLE
Dependency org.apache.tomcat:tomcat-catalina, leading to CVE problem

### ISSUE THREAD

**Issue by CVEDetect** [restx/restx#339]
Hi, In **restx-master/restx-server-tomcat**，there is a dependency **org.apache.tomcat:tomcat-catalina:7.0.47** that calls the risk method.

[CVE-2017-12615](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-12615)

The scope of this CVE affected version is **[9.0.0.M1, 9.0.30),[8.5.0,8.5.50),[,7.0.99)**

After further analysis, in this project, the main Api called is **<org.apache.naming.resources.FileDirContext: void setDocBase(java.lang.String)>**

Risk method repair link : [GitHub](https://github.com/apache/tomcat/commit/07dc0ea2745f0afab6415f22b16a29f1c6de5727)

**CVE Bug Invocation Path--**

**Path Length : 6**

```
<org.apache.naming.resources.FileDirContext: void setDocBase(java.lang.String)>
at <org.apache.catalina.core.StandardContext: boolean resourcesStart()> (org.apache.catalina.core.StandardContext.java:[5055]) in /.m2/repository/org/apache/tomcat/tomcat-catalina/7.0.47/tomcat-catalina-7.0.47.jar
at <org.apache.catalina.core.StandardContext: void startInternal()> (org.apache.catalina.core.StandardContext.java:[5235]) in /.m2/repository/org/apache/tomcat/tomcat-catalina/7.0.47/tomcat-catalina-7.0.47.jar
at <org.apache.catalina.util.LifecycleBase: void start()> (org.apache.catalina.util.LifecycleBase.java:[150]) in /.m2/repository/org/apache/tomcat/tomcat-catalina/7.0.47/tomcat-catalina-7.0.47.jar
at <org.apache.catalina.startup.Tomcat: void start()> (org.apache.catalina.startup.Tomcat.java:[341]) in /.m2/repository/org/apache/tomcat/tomcat-catalina/7.0.47/tomcat-catalina-7.0.47.jar
at <restx.server.TomcatWebServer: void _start()> (restx.server.TomcatWebServer.java:[49]) in /detect/unzip/restx-master/restx-server-tomcat/target/classes


```

**Dependency tree--**

```
[INFO] io.restx:restx-server-tomcat:jar:0.36-SNAPSHOT
[INFO] +- io.restx:restx-factory:jar:0.36-SNAPSHOT:compile
[INFO] |  +- io.restx:restx-common:jar:0.36-SNAPSHOT:compile
[INFO] |  |  \- com.samskivert:jmustache:jar:1.15:compile
[INFO] |  +- javax.inject:javax.inject:jar:1:compile
[INFO] |  +- com.google.guava:guava:jar:28.2-jre:compile
[INFO] |  |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  |  +- org.checkerframework:checker-qual:jar:2.10.0:compile
[INFO] |  |  +- com.google.errorprone:error_prone_annotations:jar:2.3.4:compile
[INFO] |  |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] |  +- joda-time:joda-time:jar:2.10.5:compile
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.30:compile
[INFO] |  \- org.reflections:reflections:jar:0.9.9-RC1:compile
[INFO] |     +- org.javassist:javassist:jar:3.16.1-GA:compile
[INFO] |     \- dom4j:dom4j:jar:1.6.1:compile
[INFO] |        \- xml-apis:xml-apis:jar:1.0.b2:compile
[INFO] +- io.restx:restx-core:jar:0.36-SNAPSHOT:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-core:jar:2.10.1:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.10.1:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.10.1:compile
[INFO] |  +- com.fasterxml.jackson.datatype:jackson-datatype-joda:jar:2.10.1:compile
[INFO] |  +- com.fasterxml.jackson.datatype:jackson-datatype-guava:jar:2.10.1:compile
[INFO] |  +- io.restx:restx-classloader:jar:0.36-SNAPSHOT:compile
[INFO] |  +- io.restx:restx-apidocs-doclet:jar:0.36-SNAPSHOT:compile
[INFO] |  |  \- com.sun.tools:tools:jar:1.8.0_292:system
[INFO] |  +- javax.validation:validation-api:jar:1.1.0.Final:compile
[INFO] |  \- org.yaml:snakeyaml:jar:1.25:compile
[INFO] +- org.apache.tomcat:tomcat-catalina:jar:7.0.47:compile
[INFO] |  +- org.apache.tomcat:tomcat-servlet-api:jar:7.0.47:compile
[INFO] |  +- org.apache.tomcat:tomcat-juli:jar:7.0.47:compile
[INFO] |  +- org.apache.tomcat:tomcat-annotations-api:jar:7.0.47:compile
[INFO] |  +- org.apache.tomcat:tomcat-api:jar:7.0.47:compile
[INFO] |  \- org.apache.tomcat:tomcat-util:jar:7.0.47:compile
[INFO] +- org.apache.tomcat.embed:tomcat-embed-core:jar:7.0.47:compile
[INFO] +- org.apache.tomcat:tomcat-jasper:jar:7.0.47:compile
[INFO] |  +- org.apache.tomcat:tomcat-jsp-api:jar:7.0.47:compile
[INFO] |  +- org.apache.tomcat:tomcat-el-api:jar:7.0.47:compile
[INFO] |  +- org.eclipse.jdt.core.compiler:ecj:jar:4.2.2:compile
[INFO] |  \- org.apache.tomcat:tomcat-jasper-el:jar:7.0.47:compile
[INFO] \- javax.servlet:javax.servlet-api:jar:3.0.1:provided
```

**_Suggested solutions:_**

Update dependency version

Thank you very much.

**Comment by CVEDetect** [comment]
@xhanin 
Could please help me check this issue?
May I pull a request to fix it?
Thanks again.

**Comment by fhebuterne** [comment]
Hi,

We have fixed it in latest restx release (1.0.0-rc1 and 1.0.0-rc2) by upgrading tomcat in latest 10.0

We plan to upgrade tomcat to 10.1 when we upgraded servlet-api to 6.0, we waiting for final release of Jetty 12 has servlet 6.0 compatibility.
